activerecord/lib/active_record/inheritance.rb

```diff
@@ -217,7 +217,7 @@ def type_condition(table = arel_table)
         def subclass_from_attributes(attrs)
           attrs = attrs.to_h if attrs.respond_to?(:permitted?)
           if attrs.is_a?(Hash)
-            subclass_name = attrs.with_indifferent_access[inheritance_column]
+            subclass_name = attrs[inheritance_column] || attrs[inheritance_column.to_sym]
```

Comment:
Yes, the default value of `inheritance_column` is `"type"`, and when the default is overridden, [the setter ensures the custom value is a string](https://github.com/rails/rails/blob/v5.1.1/activerecord/lib/active_record/model_schema.rb#L276-L279).